pkg/ingester/user_metrics_metadata_test.go

```diff
@@ -107,15 +108,132 @@ func TestUserMetricsMetadata(t *testing.T) {
 			}
 
 			// Verify expected elements are stored
-			clientMeta := mm.toClientMetadata()
+			clientMeta := mm.toClientMetadata(nil)
 			assert.ElementsMatch(t, testData.expectedMetadata, clientMeta)
 
 			// Purge all metadata
 			mm.purge(time.Time{})
 
 			// Verify all metadata purged
-			clientMeta = mm.toClientMetadata()
+			clientMeta = mm.toClientMetadata(nil)
 			assert.Empty(t, clientMeta)
 		})
 	}
 }
+
+type dummytest struct{}
+
+func (t dummytest) Errorf(string, ...interface{}) {}
+
+func TestUserMetricsMetadataOptions(t *testing.T) {
```

Comment:
This test seems very similar to `TestUserMetricsMetadata`. Did you consider modifying that one to be able to specify requests in its `tests`?

Reply:
Yes, but the focus and the needed code for both seems different enough that I opted to separate them. The new test is for options defined when calling the API, and focuses on how the metadata is queried, changing the response (with different possible combinations), while the old test is for limits in Mimir itself, focuses on how the metadata is stored/purged and checks for specific errors.